### DISCUSSION TITLE
Select connected components of the first Z object found

### DISCUSSION THREAD

**Discussion by wikyburns** [cnr-isti-vclab/PyMeshLab#84] [Q&A]
Hi, 

I was trying to delete everthing around and scan.

The idea is to select the first triangle with the minimum Z and and select the connected component around it, then invert the selection and delete vertices.

It work in the UI meshlab, but how I can do it on pymeshlab?

Here the image how i did that on the UI meshlab.

1- select the first with the manual selection "select connected components in a region:
![image](https://user-images.githubusercontent.com/11525781/111837303-fe51b780-88f7-11eb-889f-7d38adf5eadc.png)

2- invert selection (key i)
![image](https://user-images.githubusercontent.com/11525781/111837339-0c073d00-88f8-11eb-8f3a-da44e39dde18.png)

3- Delete the current set of selection
![image](https://user-images.githubusercontent.com/11525781/111837436-293c0b80-88f8-11eb-98f0-6e44cc6760a2.png)

How I can do it on pymeshlab?

Thanks

**Comment by alemuntoni** [comment]
the second and third tasks are available also in pymeshlab, just check in the documentation the corresponding names of the filters.
For the first task, you could use the "conditional face selection", giving as parameter the id of the face having with minimum Z. then use the filter "select connected faces".
